src/rrd_xport.c

```diff
@@ -947,7 +948,27 @@ static int rrd_xport_format_addprints(int flags,stringbuffer_t *buffer,image_des
 	  ("bad format for PRINT in \"%s'", im->gdes[i].format);
 	return -1;
       } else {
-        rrd_snprintf(dbuf,sizeof(dbuf), im->gdes[i].format, printval,"");
+				 char *percent_s;
+				 /* unit autoscaling for %S and %s, logic copied from rrd_graph.c */
+				 if ((percent_s = strstr(im->gdes[i].format, "%S")) != NULL) {
+						/* Magfact is set to -1 upon function entry.  If it
+						 * is still less than 0, then we need to run auto_scale.
+						 * Otherwise, put the value into the correct units.  If
+						 * the value is 0, then do not set the symbol or magnification
+						 * so next the calculation will be performed again. */
+						if (magfact < 0.0) {
+							 auto_scale(im, &printval, &si_symb, &magfact);
+							 if (printval == 0.0)
+									magfact = -1.0;
+						} else {
+							 printval /= magfact;
+						}
+						*(++percent_s) = 's';
+				 } else if (strstr(im->gdes[i].format, "%s") != NULL) {
+						auto_scale(im, &printval, &si_symb, &magfact);
+				 }
+
+        rrd_snprintf(dbuf,sizeof(dbuf), im->gdes[i].format, printval, si_symb);
       }
```

Comment:
something seems odd with the indentation here ... best use spaces

Reply:
Hi Toby,

I was going to rectify this file, but I don't know how to fix it because the indentation in this file does not follow a consistent pattern.

Reply:
don't get hung up on the indentation ... that is ok ...